src/pages/SignUp/SignUp.js

```diff
@@ -1,7 +1,214 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { SIGNUP_INPUT_LIST } from './signupInPutList';
+import './SignUp.scss';
 
 const SignUp = () => {
-  return <div>SignUp</div>;
+  const navigate = useNavigate();
+  const [inputValue, setInputValue] = useState({
+    email: '',
+    password: '',
+    passwordAgain: '',
+    name: '',
+    phoneNumber: '',
+    birthday: '',
+    adress: '',
+  });
+  const {
+    email,
+    password,
+    passwordAgain,
+    name,
+    phoneNumber,
+    birthday,
+    adress,
+  } = inputValue;
+
+  const handleInput = e => {
+    const { name, value } = e.target;
+    setInputValue({ ...inputValue, [name]: value });
+  };
+
+  const [isButtonWork, setIsButtonWork] = useState({
+    agree: false,
+    userInfo: false,
+    age: false,
+  });
+  const { agree, userInfo, age } = isButtonWork;
+
+  const isAllChecked = Object.values(isButtonWork).every(
+    value => value === true
+  );
+
+  const handleCheckBox = e => {
+    setIsButtonWork(prev => ({
+      ...prev,
+      [e.target.name]: !prev[e.target.name],
+    }));
+  };
+
+  const handleAllCheck = () => {
+    const checkboxKeys = Object.keys(isButtonWork);
+    checkboxKeys.forEach(list => {
+      return setIsButtonWork(prev => ({ ...prev, [list]: !isAllChecked }));
+    });
+  };
+
+  const conditions = {
+    emailCheck:
+      !email ||
+      new RegExp(
+        /^[0-9a-zA-Z]([-_\.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_\.]?[0-9a-zA-Z])*\.[a-zA-Z]{2,3}$/
+      ).test(email),
+
+    passwordCheck:
+      !password ||
+      new RegExp(/^(?=.*[A-Za-z])(?=.*[0-9])(?=.*[!@#$%^&*])(?=.{8,20})/).test(
+        password
+      ),
+    passwordAgainCheck: !passwordAgain || password === passwordAgain,
+    namecheck: name.length >= 2,
+    phoneNumberCheak: phoneNumber.length >= 10 && phoneNumber.length <= 12,
+    birthdayCheck: birthday.length === 6,
+    adressCheck: adress.length > 8,
+  };
+
+  const activeButton =
+    Object.values(conditions).every(value => value === true) && isAllChecked;
+
+  const goToMain = () => {
+    if (activeButton) {
+      fetch('http://10.58.52.73:8000/auth/signin', {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json;charset=utf-8' },
+        body: JSON.stringify({
+          email: email,
+          password: password,
+          passwordAgain: passwordAgain,
+          name: name,
+          phoneNumber: phoneNumber,
+          birthday: birthday,
+          adress: adress,
+        }),
+      }).then(response => response.json());
+      alert('가입이 완료되었습니다. 즐거운 쇼핑 되세요!');
+    } else {
+      alert('필수 정보를 정확하게 입력 해주세요!');
+    }
+  };
+
+  return (
+    <div className="SignUp">
```

Comment:
```suggestion
    <div className="signUp">
```
className에는 소문자로 시작하도록 해주세요!